fn/compare.xml

```diff
@@ -1606,4 +1930,44 @@ return compare($a, $b)
          <assert-eq>1</assert-eq>
       </result>
    </test-case>
+   
+   <test-case name="compare-error-01">
+      <description>compare xs:QNames</description>
+      <created by="Michael Kay" on="2024-03-23"/>
+      <dependency type="spec" value="XP40+ XQ40+"/>
+      <test>compare(xs:QName('one'), xs:QName('two'))</test>
+      <result>
+         <error code="XPTY0004"/>
+      </result>
+   </test-case>
+   
+   <test-case name="compare-error-02">
+      <description>compare integer to string</description>
+      <created by="Michael Kay" on="2024-03-23"/>
+      <dependency type="spec" value="XP40+ XQ40+"/>
+      <test>compare(1, "fred")</test>
+      <result>
+         <error code="XPTY0004"/>
+      </result>
+   </test-case>
+   
+   <test-case name="compare-error-03">
+      <description>compare xs:duration values</description>
+      <created by="Michael Kay" on="2024-03-23"/>
+      <dependency type="spec" value="XP40+ XQ40+"/>
+      <test>compare(xs:duration('P1Y'), xs:duration('P12M'))</test>
+      <result>
+         <error code="XPTY0004"/>
+      </result>
+   </test-case>
+   
+   <test-case name="compare-error-04">
+      <description>compare xs:dateTime values with and without timezone</description>
+      <created by="Michael Kay" on="2024-03-23"/>
+      <dependency type="spec" value="XP40+ XQ40+"/>
+      <test>compare(current-dateTime, xs:dateTime('2024-03-23T12:00:00'))</test>
```

Comment:
```suggestion
      <test>compare(current-dateTime(), xs:dateTime('2024-03-23T12:00:00'))</test>
```

I would expect `1` instead of `XPTY0004`. Doesn’t `current-dateTime() ge xs:dateTime('2024-03-23T12:00:00')` return `true`?

Reply:
I have made the change as requested.